File: app/src/main/java/dk/eightyplus/annoto/Callback.java
package dk.eightyplus.annoto;

import dk.eightyplus.annoto.action.State;
import dk.eightyplus.annoto.action.Undo;
import dk.eightyplus.annoto.component.Component;

/**
 *
 */
public interface Callback {

  public void textEditDone();

  public void move(Component component, float dx, float dy, float scale);

  public State getState();

  public void setState(State state);

  public void setStrokeWidth(int width);

  public void add(Undo undo);

  public void load(String fileName);

  public void colorChanged(int color);
}

File: app/src/main/java/dk/eightyplus/annoto/Callback.kt
package dk.eightyplus.annoto

import dk.eightyplus.annoto.action.State
import dk.eightyplus.annoto.action.Undo
import dk.eightyplus.annoto.component.Component

/**
 *
 */
interface Callback {

    var state: State

    fun textEditDone()

    fun move(component: Component, dx: Float, dy: Float, scale: Float)

    fun setStrokeWidth(width: Int)

    fun add(undo: Undo)

    fun load(fileName: String)

    fun colorChanged(color: Int)

    fun hexColorForResourceId(color: Int): Int
}
